### PR TITLE
Fixed Bittrex syncing and parsing and updated unit tests

### PR DESCRIPTION
**Pull request by einsteinx2**
**Is this pull request adding a feature or fixing a bug?**  
bug

**Does this pull request close an issue? If so, which one?**
Bittrex doesn't work (bad parsing and BCC prices) #507

**What does this pull request do? What does it change?**
Fixed Bittrex syncing and parsing and updated unit tests, also fixed special case where they use the wrong symbol (BCC) for Bitcoin Cash (which should be BCH).

### CODE REVIEW

File: Balance/Shared/Data Model/Syncing/Syncer.swift [hunk 1]
```diff
@@ -431,33 +431,53 @@ class Syncer {
...
+                    guard let date = deposit.date else {
+                        log.error("Failed to format date for Bittrex deposit \(deposit.id) with date string \(deposit.lastUpdated)")
                         continue
                     }
                     
-                    let amount = paddedInteger(for: transaction.amount, currencyCode: transaction.currency)
+                    let amount = paddedInteger(for: deposit.amount, currencyCode: deposit.currency)
                     
-                    TransactionRepository.si.transaction(source: institution.source, sourceTransactionId: transaction.paymentUuid, sourceAccountId: transaction.currency, name: transaction.paymentUuid, currency: transaction.currency, amount: amount, date: date, categoryID: nil, institution: institution)
+                    // NOTE: Maybe we shoul be using txId here instead of id?
```
Comment (wolffan) [inline Balance/Shared/Data Model/Syncing/Syncer.swift]: agree with the comment

Comment (einsteinx2) [reply Balance/Shared/Data Model/Syncing/Syncer.swift]: The thing is id is technically the actual source transaction id (i.e. the id for the transaction in Bittrex), but I agree txId is more useful, but I feel like we should have a special field for that. In any case, it's more useful to use now, so I'll make the change.

**Comment by einsteinx2** [comment]
Also the backend work is done, we now have prices from Bittrex including the BCC to BCH fix.

**Comment by einsteinx2** [comment]
I tested using the test credentials from @jw

File: Balance/Shared/Data Model/Syncing/Syncer.swift (hunk 1 above)
Comment (einsteinx2) [reply Balance/Shared/Data Model/Syncing/Syncer.swift]: Oh actually never mind, the other problem is that txId is not guaranteed to be returned, so we can't rely on it for this. We should add special transaction Id handling later.

**Pull request merged**